substrate/console

```diff
@@ -218,10 +220,10 @@ namespace substrate
 			if (value_ < 0)
 			{
 				stream.write('-');
-				format(stream, uint_t(uint_t(~uint_t(value_)) + 1U));
+				format(stream, ~static_cast<uint_t>(value_) + 1U);
```

Comment:
Typo?

Reply:
Shouldn't be - the old code was trying to compensate for `uint_t` being an unsigned integral type that would promote repeatedly - bitwise not resulting in `int`, adding one resulting in `int`, and the original initial cast of the value to an unsigned type.

With `uint_t` being a non-promoting type thanks to the promoted_type_t helper, the only cast that should be required is that initial cast. The rest of the expression is a manual implementation of the unary `operator -` because `-uint_t(value_)` results in compiler warnings too, and this path needs to turn negative numbers positive.

Reply:
Just realised that.. we did screw one thing up - unless the compiler sign-extends `value_` before the cast, the bitwise not should result in something like `0xff` -> `0xffffff00` happening which is not what's intended here